core/src/com/mygdx/battleship/BattleshipBots/SmartRandomBot.java

```diff
@@ -0,0 +1,244 @@
+                            placement[x] = newPlace;
+                        }
+                    }
+                    break;
+
+                case EAST:
+                    for (int x = 1; x < shipSizes[curShip]; x++) {
+                        char newL = (char) (startL + x);
+                        String newPlace = String.valueOf(Character.toUpperCase(newL)) + startR;
+                        if (usedCoordinates.contains(newPlace) || !BattleshipUtils.validateCoordinate(newPlace)) {
+                            reset = true;
+                            break;
+                        } else {
+                            placement[x] = newPlace;
+                        }
+                    }
+                    break;
+
+                case SOUTH:
+                    for (int x = 1; x < shipSizes[curShip]; x++) {
+                        int newR = startR - x;
+                        String newPlace = String.valueOf(Character.toUpperCase(startL)) + newR;
+                        if (usedCoordinates.contains(newPlace) || !BattleshipUtils.validateCoordinate(newPlace)) {
+                            reset = true;
+                            break;
+                        } else {
+                            placement[x] = newPlace;
+                        }
+                    }
+                    break;
+            }
+
+            //check invariants
+            reset = reset || !BattleshipUtils.validateShip(placement);
+            if (!reset) {
+                shipsLocations[curShip] = placement;
+                curShip++;
+                usedCoordinates.addAll(Arrays.asList(placement));
+            }
+        }
+    }
+
+	/**
+	 * Randomly decides next move.
+     * If hit in last turn, randomly choose a direction and continue on direction
+     * If another hit, continue on direction.
+     * If it has found a ship, will intelligently shoot around the ship until it dies.
+	 */
+	private void nextMove() {
+        if ((sink)||(!foundShip)) {
+            usedDirections.clear();
+            move = getRandomMove();
+        } else {
+            boolean badCoordinate = false;
+            while (true) {
+                //if miss, select an untried direction, otherwise keep direction
+                if ((!hit)||badCoordinate) {
+                    usedDirections.add(direction);
+                    OrientationType newDirection = OrientationType.values()[(direction.ordinal()+2)%4];
+                    Random rand = new Random();
+                    if (usedDirections.size() == 4) {
+                        usedDirections.clear();
+                        lastHits.removeLast();
+                    }
+                    while (usedDirections.contains(newDirection)) {
+                        newDirection = OrientationType.values()[rand.nextInt(4)];
+                    }
+                    direction = newDirection;
+                }
+                String lastHit = lastHits.getLast();
```

Comment:
Sorry it's a NoSuchElement exception and it happens pretty much immediately upon running the game.

Consider an execution that hits the conditional on line 147 then continues to line 156.